### DISCUSSION TITLE
Compiled package version No. starts with a v - Debian dpkg gives error - version number does not start with digit

### DISCUSSION THREAD

**Discussion by BobManMasher** [gerbera/gerbera#1225] [Q&A]
Having successfully compiled Gerbera I used cpack to package it up. The first run produced tar files so I then added the following lines to CMakeLists.txt

`SET(CPACK_GENERATOR "DEB")`
`SET(CPACK_DEBIAN_PACKAGE_MAINTAINER "KK")` 

I cleared my build folder, reran the cmake and make commands which both completed without error. cpack produced a .deb file which I tried to install with dpkg but I then got the following error:

`root@build-gerbera# dpkg -i gerbera_v1.6.4-185-gae283931+d_armel.deb`
`dpkg: error processing archive gerbera_v1.6.4-185-gae283931+d_armel.deb (--install):`
 `parsing file '/var/lib/dpkg/tmp.ci/control' near line 8 package 'gerbera':`
 `'Version' field value 'v1.6.4-185-gae283931+d': version number does not start with digit`
`Errors were encountered while processing:`
 `gerbera_v1.6.4-185-gae283931+d_armel.deb` 

Is it safe to change the version number and where is it set?

**Comment by BobManMasher** [comment]
When I  use head CMakeLists.txt I get ...
` Only set for release tags`
` You should only see version "git" when built from a git snapshot tarball`
`set(GERBERA_VERSION "git")`

I assume that the problem is caused by the version number provided by git. When I look at other projects on here, none of them have version numbers starting with v. Is that right? if not, how can I check and see if the version number is set as v1 instead of just 1?

**Reply by whyman** [reply]
In our "build-deb.sh" script we have this logic:

https://github.com/gerbera/gerbera/blob/11b9273b75973aa05c753cfdeaba4227f2c5f207/scripts/debian/build-deb.sh#L125

The last sed strips off the `v`

Then we pass a version directly into cpack:
https://github.com/gerbera/gerbera/blob/11b9273b75973aa05c753cfdeaba4227f2c5f207/scripts/debian/build-deb.sh#L145

**Reply by BobManMasher** [reply]
OK - so it seems to me that I need to specify Deb somewhere when I run the build 'make -j4' command as the cpack script section just uses the number embedded in the build process which seems to be bypassing the stripping of the leading 'v'. Can I force the cmake/make to use the build-deb.sh?

**Reply by whyman** [reply]
You can override the version used by cpack by setting it on the command line as above:

So you would run something something like:

`cpack -G DEB CPACK_DEBIAN_PACKAGE_VERSION=1.6.4-185-gae283931+d`

**Reply by BobManMasher** [reply]
So I tried the command lines as suggested but I still got the 'v' problem in the file name and content.  The only way to fix this was to edit CMakeFiles.txt and change the version from 'git' to:

`set(GERBERA_VERSION "1.6.4-185-gae283931+d")`

After the cpack I get a .deb file which installs correctly however I have the following discrepancies/problems:

1) the gerbera.service is in my build directory but not copied to /etc/systemd/system - had to copy by hand.
2) all references to files e.g.the location of gerbera, point to /usr/local/bin/gerbera which is true if you do a 'make install' but dpkg installs the files to /usr/bin/gerbera (the correct place I believe for init.d or systemd scripts). This is also true of files in /usr/local/share gerbera which get installed into /usr/share/gerbera.
3) The sqlite3 script isn't found unless I copy the files into /usr/local/share/gerbera.  I could not see a setting in config.xml to point gerbera to the installed folder which left me no option but to copy the files.
4) if I start gerbera manually, it starts correctly and I can access the server via the network. If I edit gerbera.service to point to /etc/bin/gerbera, I can start the service but it exits with an error. When I run the command as user gerbera, I get the following error:

`info: Loading configuration from: /etc/gerbera/config.xml`
`info: Checking configuration...`
`error: filesystem error: status: Permission denied [/root/.config/gerbera]`

this is easily fixed with an edit of the 'home' path in config.xml .

4) there is no init.d script file in the current build but the old one that was installed with either 1.1 or 1.3 seems to work. 

5) I found that the best place to store config.xml is in /etc/gerbera and then make a symbolic link from /etc/gerbera/config.xml to the gerbera user home directory ./home/gerbera/.config/gerbera  

I was able to take my install package and load it onto a different PogoNAS box and making these changes and loading the correct libs, was able to get it working successfully.

My final problem now is that APT doesn't recognise my new version of gerbera as being the latest and keeps trying to downgrade me to v1.3

Should I raise these as issues?
